### PR TITLE
feat: add taxJurisdictionCode field to physical address

### PR DESCRIPTION
**Pull request by nicoprow**
## Description

This pull request adds the new field 'taxJurisdictionCode' to the business partner's physical address

- changed API models
- added database column
- adapted the mappings
- enhanced the test data
- updated the api documentation

implements to #955

## Pre-review checks

Please ensure to do as many of the following checks as possible, before asking for committer review:

- [x] DEPENDENCIES are up-to-date. [Dash license tool](https://github.com/eclipse/dash-licenses). Committers can open IP issues for restricted libs.
- [x] [Copyright and license header](https://eclipse-tractusx.github.io/docs/release/trg-7/trg-7-02) are present on all affected files

### CODE REVIEW

File: docs/api/gate.json [hunk 1]
```diff
@@ -3,14 +3,22 @@
   "info": {
     "title": "Business Partner Data Management Gate",
     "description": "A gate for a member to share business partner data with CatenaX",
-    "version": "6.0.0"
+    "version": "6.0.2-SNAPSHOT"
```
Comment (SujitMBRDI) [inline docs/api/gate.json:6]: Version is updated now, need to update docs of api for  each service or we shall prefer to adapt the same during feature freeze?

**Review by SujitMBRDI — COMMENTED** [review]
All changes look good to me. Just need to make decision related to api docs version.

File: docs/api/gate.json (hunk 1 above)
Comment (nicoprow) [reply docs/api/gate.json:6]: I will update the API docs with the newest SNAPSHOT version and then we can have a final look again

Comment (SujitMBRDI) [reply docs/api/gate.json]: Yeah, we can do that.

**Review by SujitMBRDI — APPROVED** [review]

**Pull request merged**